src/app/test/test-folder-tree/test-folder-tree.component.ts

```diff
@@ -24,81 +26,59 @@ export class TestFolderTreeComponent {
+      if (item.path) {
+        this.nextAction(
+          item,
+          item.path.split('/').filter((s) => s !== ''),
+          this.rootFolder,
+        );
+      }
     }
-    return path;
+    this.tree.addItem(this.rootFolder);
   }
 
-  setData(data: CreateTreeItem[]): void {
-    this.resetTree();
-    this.originalItems = data;
-    const tempItems: CreateTreeItem[] = this.convertToFolders(data);
-    const rootFolder: CreateTreeItem = Object.assign({}, this.rootFolder);
-    if (tempItems && tempItems.length > 0) {
-      rootFolder.children = tempItems;
-    }
-    this.tree.addItem(rootFolder);
-  }
+  nextAction(itemToAdd: CreateTreeItem, routeParts: string[], currentLocation: CreateTreeItem) {
+    if (routeParts.length > 0) {
+      const nextRoutePart = routeParts[0];
+      let newChild = this.getChildForNextRoutePart(nextRoutePart, currentLocation);
+      if (currentLocation.children) {
+        currentLocation.children.sort((a, b) => a.name.localeCompare(b.name));
+      } else {
+        currentLocation.children = [];
+      }
 
-  convertToFolders(items: CreateTreeItem[]): CreateTreeItem[] {
-    const tempItems: CreateTreeItem[] = [];
-    const folderNames: string[] = [];
-    for (let item of items) {
-      if (item.path) {
-        const tempItem: CreateTreeItem | undefined = this.createFolderFromPath(item.path);
-        if (tempItem && !folderNames.includes(tempItem.path!)) {
-          folderNames.push(tempItem.path!);
-          tempItems.push(tempItem);
-        }
+      if (!newChild) {
+        newChild = {
+          name: nextRoutePart,
+        } as CreateTreeItem;
+        currentLocation.children.push(newChild);
       }
+      this.nextAction(itemToAdd, routeParts.slice(1), newChild);
     }
-    return tempItems;
   }
 
-  createFolderFromPath(path: string): CreateTreeItem | undefined {
-    let index = path.indexOf('/');
-    if (index === 0) {
-      index = path.indexOf('/', 2);
-    }
-    if (index === -1) {
-      return undefined;
-    }
-    let folderName = path.slice(0, index);
-    const remainingPath = path.replace(folderName, '');
-    if (folderName.startsWith('/')) {
-      folderName = folderName.slice(1);
-    }
-    const treeItem: CreateTreeItem = { name: folderName };
-    if (remainingPath && remainingPath !== '/' && remainingPath !== '') {
-      const child = this.createFolderFromPath(remainingPath);
-      if (child) {
-        treeItem.children = [child];
+  getChildForNextRoutePart(routePart: string, currentLocation: CreateTreeItem) {
+    if (currentLocation.children) {
+      for (const child of currentLocation.children) {
+        if (child.name === routePart) {
+          return child;
+        }
       }
     }
-    return treeItem;
+    return null;
   }
 
-  resetTree(): void {
-    if (this.tree) {
-      this.tree.clearItems();
-    }
-    this.originalItems = [];
+  autoSelectItem(item: CreateTreeItem): boolean {
+    return item.path === this.rootFolder.name;
   }
```

Comment:
This is only used in `autoSelectCondition` right?
Can't you do it inline in there since you don't use it anywhere else?

Reply:
Fixed